Tests/SentryTests/Transaction/SentryTracerTests.swift

```diff
@@ -1179,30 +1182,35 @@ class SentryTracerTests: XCTestCase {
     func testConcurrentTransactions_OnlyOneGetsMeasurement() {
         SentrySDKInternal.setAppStartMeasurement(fixture.getAppStartMeasurement(type: .warm))
         
-        let queue = DispatchQueue(label: "", qos: .background, attributes: [.concurrent, .initiallyInactive] )
-        let group = DispatchGroup()
-        
+        let queue = DispatchQueue(label: "testConcurrentTransactions_OnlyOneGetsMeasurement", attributes: [.concurrent, .initiallyInactive] )
+
         let transactions = 5
+        let finishTransactionExpectation = XCTestExpectation(description: "Finish transactions")
+        finishTransactionExpectation.expectedFulfillmentCount = transactions
+
         for _ in 0..<transactions {
-            group.enter()
             queue.async {
-                self.fixture.getSut().finish()
-                group.leave()
+                let tracer = self.fixture.getSut()
+
+                tracer.finish()
+                finishTransactionExpectation.fulfill()
             }
         }
         
         queue.activate()
-        group.wait()
-        
-        XCTAssertEqual(transactions, fixture.hub.capturedEventsWithScopes.count)
-        
+
+        wait(for: [finishTransactionExpectation], timeout: 10.0)
```

Comment:
`l`: Is changing the approach from a dispatch group to a waiting queue the same test case? or are we testing two different cases now. I'm fine with both, but please add a short comment so we remember when touching this again, why we chose XCTestExpectation > DispatchGroup

Reply:
I plan on replacing all DispatchGroups with XCTestExpectation in all tests, see https://github.com/getsentry/sentry-cocoa/issues/5789. I will also add a linter to explain this for all test cases if somebody again adds a DispatchGroup in a test.